File: src/tonemap/IdentityToneMap.cpp
#include "tonemap/IdentityToneMap.h"

IdentityToneMap::IdentityToneMap() {
}

IdentityToneMap::~IdentityToneMap() {
}

void
IdentityToneMap::init() {
}

ColorRgb
IdentityToneMap::scaleForComputations(ColorRgb radiance) const {
    return radiance;
}

ColorRgb
IdentityToneMap::scaleForDisplay(ColorRgb radiance) const {
    return radiance;
}

File: java/src/main/java/vsdk/toolkit/tonemap/IdentityToneMap.java
package vsdk.toolkit.tonemap;

import vsdk.toolkit.common.ColorRgb;

public final class IdentityToneMap extends ToneMap {
    public IdentityToneMap() {
    }

    /*toneMapOptions*/
    @Override
    public void init(ToneMappingContext toneMapOptions) {
    }

    @Override
    public ColorRgb scaleForComputations(ColorRgb radiance) {
        return radiance;
    }

    @Override
    public ColorRgb scaleForDisplay(ColorRgb radiance) {
        return radiance;
    }
}
